src/main/java/org/broadinstitute/hellbender/tools/variantdb/arrays/RawArrayFieldEnum.java

```diff
@@ -3,126 +3,96 @@
-            String lrr = LRR.getColumnValue(variant, probeInfo, sampleId);
+    sample_id,
+    probe_id,
+    GT_encoded { // Required
+        public String getColumnValue(final VariantContext variant) {
+            List<Integer> alleleIndexes = CommonCode.getGTAlleleIndexes(variant);
 
-            RawArrayData d = new RawArrayData(convert(normx),
-                                              convert(normy),
-                                              convert(baf),
-                                              convert(lrr)
-                                              );
-            return String.valueOf(d.encode());
-        }
-    },
-
-    probe_id { // Required
-        public String getColumnValue(final VariantContext variant, ProbeInfo probeInfo, String sampleId) {
-            return String.valueOf(probeInfo.probeId);
-        }
-    },
-
-    GT_encoded {
-        public String getColumnValue(final VariantContext variant, ProbeInfo probeInfo, String sampleId) {
-            Genotype g = variant.getGenotype(0);
             RawArrayTsvCreator.GT_encoding gt = RawArrayTsvCreator.GT_encoding.MISSING;
-            if (g.isHomRef() || g.isHomVar()) {
-                Allele allele = g.getAllele(0);
-                if (allele.basesMatch(probeInfo.alleleA)) {
-                    gt = RawArrayTsvCreator.GT_encoding.AA;
-                } else if (allele.basesMatch(probeInfo.alleleB)) {
-                    gt = RawArrayTsvCreator.GT_encoding.BB;
+            if (alleleIndexes.size() == 2) {
+                Set<Integer> uniqueAlleleIndexes = new HashSet<>(alleleIndexes);
+
+                if (uniqueAlleleIndexes.size() == 1) {
+                    // we know it's HOM something
+                    if (uniqueAlleleIndexes.contains(0)) {
+                        gt = RawArrayTsvCreator.GT_encoding.HOM_REF;
+                    } else if (uniqueAlleleIndexes.contains(1)) {
+                        gt = RawArrayTsvCreator.GT_encoding.HOM_VAR;
+                    } else if (uniqueAlleleIndexes.contains(2)) {
+                        gt = RawArrayTsvCreator.GT_encoding.HOM_ALT2;
+                    }
                 } else {
-                    throw new IllegalStateException("allele: " + allele + " must match either A: " + probeInfo.alleleA + " or B: " + probeInfo.alleleB);
+                    // we know its het
+                    if (uniqueAlleleIndexes.containsAll(new HashSet<>(Arrays.asList(0, 1)))) {
+                        gt = RawArrayTsvCreator.GT_encoding.HET0_1;
+                    } else if (uniqueAlleleIndexes.containsAll(new HashSet<>(Arrays.asList(1, 2))))
+                        gt = RawArrayTsvCreator.GT_encoding.HET1_2;
                 }
-            } else if (g.isHet()) {
-                gt = RawArrayTsvCreator.GT_encoding.AB;
             }
-            return gt.getValue();
+            return gt == RawArrayTsvCreator.value_to_drop ? "null" : gt.getValue();
```

Comment:
do you mean the string `null` or an actual null?

Reply:
yes, the actual string "null". the tsv needs some value and the bq import will convert it to an actual null in the db. added to the comment to explain